### PR TITLE
BUG: conversion a JSON field descriptor into pandas type for deprecated offsets frequency 'M'

### PR DESCRIPTION
**Pull request by natmokval**
xref #52064

when converting a JSON field descriptor into its corresponding pandas type we don't check if we have deprecated for offsets frequency "M".
corrected the definition of `convert_json_field_to_pandas_type`, added a test.

### CODE REVIEW

File: pandas/io/json/_table_schema.py [hunk 1]
```diff
@@ -207,6 +208,8 @@ def convert_json_field_to_pandas_type(field) -> str | CategoricalDtype:
         if field.get("tz"):
             return f"datetime64[ns, {field['tz']}]"
         elif field.get("freq"):
+            # GH#9586 rename frequency M to ME for offsets
+            field["freq"] = freq_to_period_freqstr(1, field["freq"])
```
Comment (MarcoGorelli) [inline pandas/io/json/_table_schema.py:212]: what if `field['freq']` is `'2M'`?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for noticing and fixing! just got a question

File: pandas/io/json/_table_schema.py (hunk 1 above)
Comment (natmokval) [reply pandas/io/json/_table_schema.py]: thanks, you are right, it doesn't work for '2M'. I corrected the definition of `convert_json_field_to_pandas_type` and updated my test.

File: pandas/tests/io/json/test_json_table_schema.py [hunk 2]
```diff
@@ -845,3 +845,13 @@ def test_read_json_orient_table_old_schema_version(self):
         expected = DataFrame({"a": [1, 2.0, "s"]})
         result = pd.read_json(StringIO(df_json), orient="table")
         tm.assert_frame_equal(expected, result)
+
+    def test_read_json_table_orient_period_depr_freq(self, recwarn):
+        # GH#9586
+        df = DataFrame(
+            {"ints": [1, 2]},
+            index=pd.PeriodIndex(["2011-01", "2011-08"], freq="2M"),
```
Comment (MarcoGorelli) [inline pandas/tests/io/json/test_json_table_schema.py:853]: could we check it works for both `'2M'` and `'M'`?

Comment (natmokval) [reply pandas/tests/io/json/test_json_table_schema.py]: yeah, sure, I added parameterization to the test, for `'M'` and `'2M'`.

File: pandas/io/json/_table_schema.py [hunk 3]
```diff
@@ -207,8 +209,12 @@ def convert_json_field_to_pandas_type(field) -> str | CategoricalDtype:
         if field.get("tz"):
             return f"datetime64[ns, {field['tz']}]"
         elif field.get("freq"):
+            # GH#9586 rename frequency M to ME for offsets
+            freq_name = re.split("[0-9]*", field["freq"], maxsplit=1)[1]
+            freq_n = field["freq"][: field["freq"].index(freq_name)]
+            freq = freq_to_period_freqstr(freq_n, freq_name)
```
Comment (MarcoGorelli) [inline pandas/io/json/_table_schema.py:215]: having a regular expression here looks a bit complex - would it work to do
```python
offset = to_offset(field['freq'])
freq_n, freq_name = offset.n, offset.name
freq = freq_to_period_freqstr(freq_n, freq_name)
```
instead?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/io/json/_table_schema.py (hunk 3 above)
Comment (natmokval) [reply pandas/io/json/_table_schema.py]: thank you, it works very well. I replaced the regular expression with to_offset and updated my PR.

**Review by MarcoGorelli — APPROVED** [review]
nice one, thanks @natmokval !

**Comment by natmokval** [comment]
Thank you @MarcoGorelli for reviewing this PR.

**Pull request merged**